packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -253,12 +270,12 @@ class _CheckboxState extends State<Checkbox> with TickerProviderStateMixin {
     final BoxConstraints additionalConstraints = BoxConstraints.tight(size);
     final MouseCursor effectiveMouseCursor = MaterialStateProperty.resolveAs<MouseCursor>(
       widget.mouseCursor ?? MaterialStateMouseCursor.clickable,
-      <MaterialState>{
-        if (!enabled) MaterialState.disabled,
-        if (_hovering) MaterialState.hovered,
-        if (_focused) MaterialState.focused,
-        if (widget.tristate || widget.value!) MaterialState.selected,
-      },
+      _states,
+    );
+    final Color activeColor = widget.activeColor ?? themeData!.toggleableActiveColor;
```

Comment:
This isn't quite right. If widget.activeColor is non-null, it can still resolve to null. If that happens then the theme color should be used.